pkg/manager/member/pd_ms_scaler.go

```diff
@@ -0,0 +1,131 @@
+// Copyright 2023 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package member
+
+import (
+	"fmt"
+
+	"github.com/pingcap/tidb-operator/pkg/apis/pingcap/v1alpha1"
+	"github.com/pingcap/tidb-operator/pkg/controller"
+	"github.com/pingcap/tidb-operator/pkg/util"
+	apps "k8s.io/api/apps/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/klog/v2"
+)
+
+type pdMSScaler struct {
+	generalScaler
+}
+
+// NewPDMSScaler returns a PD Micro Service Scaler.
+func NewPDMSScaler(deps *controller.Dependencies) *pdMSScaler {
+	return &pdMSScaler{generalScaler: generalScaler{deps: deps}}
+}
+
+// Scale scales in or out of the statefulset.
+func (s *pdMSScaler) Scale(meta metav1.Object, oldSet *apps.StatefulSet, newSet *apps.StatefulSet) error {
+	scaling, _, _, _ := scaleOne(oldSet, newSet)
+	if scaling > 0 {
+		return s.ScaleOut(meta, oldSet, newSet)
+	} else if scaling < 0 {
+		return s.ScaleIn(meta, oldSet, newSet)
+	}
+	return nil
+}
+
+// ScaleOut scales out of the statefulset.
+func (s *pdMSScaler) ScaleOut(meta metav1.Object, oldSet *apps.StatefulSet, newSet *apps.StatefulSet) error {
+	_, ordinal, replicas, deleteSlots := scaleOne(oldSet, newSet)
+	resetReplicas(newSet, oldSet)
+	obj, ok := meta.(runtime.Object)
+	if !ok {
+		klog.Errorf("cluster[%s/%s] can't convert to runtime.Object", meta.GetNamespace(), meta.GetName())
+		return nil
+	}
+	serviceName := controller.PDMSTrimName(oldSet.Name)
+	skipReason, err := s.deleteDeferDeletingPVC(obj, v1alpha1.PDMSMemberType(serviceName), ordinal)
```

Comment:
Do we need to support PVC for PDMS? if not, can we disable this with some check?